tests/test-integration/src/test/java/org/eclipse/jetty/test/AliasCheckerMultipleResourceBasesTest.java

```diff
@@ -0,0 +1,161 @@
+import org.eclipse.jetty.util.resource.PathResource;
+import org.eclipse.jetty.util.resource.Resource;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class AliasCheckerMultipleResourceBasesTest
+{
+    private static Server _server;
+    private static ServerConnector _connector;
+    private static HttpClient _client;
+    private static ServletContextHandler _context;
+    private static Path _webRootPath;
+    private static Path _altDir1Symlink;
+    private static Path _altDir2Symlink;
+
+    private static Path getResource(String path) throws Exception
+    {
+        URL url = AliasCheckerMultipleResourceBasesTest.class.getClassLoader().getResource(path);
+        assertNotNull(url);
+        return new File(url.toURI()).toPath();
+    }
+
+    private static void delete(Path path)
+    {
+        IO.delete(path.toFile());
+    }
+
+    private static void setAliasCheckers(ContextHandler.AliasCheck... aliasChecks)
+    {
+        _context.clearAliasChecks();
+        if (aliasChecks != null)
+        {
+            for (ContextHandler.AliasCheck aliasCheck : aliasChecks)
+            {
+                _context.addAliasCheck(aliasCheck);
+            }
+        }
+    }
+
+    @BeforeAll
+    public static void beforeAll() throws Exception
+    {
+        _webRootPath = getResource("webroot");
+
+        _altDir1Symlink = _webRootPath.resolve("../altDir1Symlink");
+        delete(_altDir1Symlink);
+        Path altDir1 = _webRootPath.resolve("../altDir1").toAbsolutePath();
+        Files.createSymbolicLink(_altDir1Symlink, altDir1).toFile().deleteOnExit();
+
+        _altDir2Symlink = _webRootPath.resolve("../altDir2Symlink");
+        delete(_altDir2Symlink);
+        Path altDir2 = _webRootPath.resolve("../altDir2").toAbsolutePath();
+        Files.createSymbolicLink(_altDir2Symlink, altDir2).toFile().deleteOnExit();
+
+        // Create and start Server and Client.
+        _server = new Server();
+        _connector = new ServerConnector(_server);
+        _server.addConnector(_connector);
+        _context = new ServletContextHandler();
+
+        _context.setContextPath("/");
+        _context.setBaseResource(new PathResource(_webRootPath));
+        _context.setWelcomeFiles(new String[]{"index.html"});
+        _context.getMimeTypes().addMimeMapping("txt", "text/plain;charset=utf-8");
+        _server.setHandler(_context);
+        _context.clearAliasChecks();
+
+        _client = new HttpClient();
+        _client.start();
+    }
+
+    @AfterAll
+    public static void afterAll() throws Exception
+    {
+        Files.delete(_altDir1Symlink);
+        Files.delete(_altDir2Symlink);
+
+        _client.stop();
+        _server.stop();
+    }
+
+    @Test
+    public void test() throws Exception
```

Comment:
this test will only work on FSs with symlink support, so you probably need some assumes

Reply:
Most of our tests with symlinks don't have any assumes at all.

Windows has supported symlinks for a long time now so don't think its needed anyway.